### PR TITLE
Streaming band subsetting

### PR DESCRIPTION
**Pull request by lossyrob**
This PR is aimed at allowing subsets of bands to be pulled from GeoTiffs in a way that avoids reading undesired bands in the case that the GeoTiffs is stored with band interleave.

### CODE REVIEW

File: raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala [hunk 1]
```diff
@@ -29,6 +29,11 @@ class LazySegmentBytes(
 ) extends SegmentBytes with LazyLogging {
   import LazySegmentBytes.Segment
 
+  // TODO: make it a parameter in GeoTrellis 2.0
+  // max distance between two segments in a group
+  // used in a chunkSegments function
+  private val minChunkSize: Int = 1024
```
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala]: Is there an issue for this, or some way to track this TODO? Is it possible to use deprecation or overloads to solve this?

File: raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala [hunk 2]
```diff
@@ -65,7 +70,15 @@ class LazySegmentBytes(
...
       .sortBy(_.startOffset) // sort segments such that we inspect them in disk order
       .foldLeft((0l, List(List.empty[Segment]))) { case ((chunkSize, headChunk :: commitedChunks), seg) =>
-      if (chunkSize + seg.size <= maxChunkSize)
+      // difference of offsets should be <= minChunkSize
+      // otherwise everything between these offsets would be read by reader
+      // and the intention is to group segments by location and to limit groups by size
+      val notChunkByOffset =
+        headChunk.headOption.map { c =>
+          seg.startOffset - c.endOffset <= minChunkSize
+        }.getOrElse(true)
```
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala]: Confusing variable name. Perhaps `isSegmentNearChunk` or something else more descriptive?

File: raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala (hunk 1 above)
Comment (pomadchin) [reply raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala]: I'll try some variants, mb it's possible smth, I agree that we can forget about it

File: raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala (hunk 2 above)
Comment (pomadchin) [reply raster/src/main/scala/geotrellis/raster/io/geotiff/LazySegmentBytes.scala]: Yea, your variant looks much better

**Comment by pomadchin** [comment]
@lossyrob @echeipesh I tried to use constructor overloads https://github.com/locationtech/geotrellis/pull/2115/commits/e4f4ca511afb8cf7aeba1a14fc280e3845434317 but it still looks like an API change. The main problem is that we can't overload functions with default values, i'd be glad if we could use just the following thing: 

```scala
class LazySegmentBytes(
  byteReader: ByteReader,
  tiffTags: TiffTags,
  maxChunkSize: Int = 32 * 1024 * 1024,
  maxOffsetBetweenChunks: Int = 1024
) extends SegmentBytes with LazyLogging { ... }
```

**Comment by pomadchin** [comment]
I think, as we changed the milestone, i can remove constructor overloads and use default methods.

**Pull request merged**